sklearn/tree/export.py

```diff
@@ -781,3 +782,173 @@ def export_graphviz(decision_tree, out_file=None, max_depth=None,
     finally:
         if own_file:
             out_file.close()
+
+
+def export_ascii(decision_tree, feature_names=None, class_names=None,
+                 max_depth=10, show_value=False, show_class=None,
+                 show_leaves_value=True, spacing=3):
+    """Build a text report showing the rules of a decision tree.
+
+    Parameters
+    ----------
+    decision_tree : decision tree
+        The decision tree to be exported.
+
+    feature_names : list, optional (default=None)
+        A list of length n_features containing the feature names.
+        If None generic names will be used ("feature_0", "feature_1", ...).
+
+    class_names : list of strings, bool or None, optional (default=None)
+        Names of each of the target classes in ascending numerical order.
+        Only relevant for classification, not supported for multi-output.
+        Ignored if show_class is False.
+
+    max_depth : int, optional (default=10)
+        Only the first max_depth levels of the tree are printed.
+
+    show_value : bool, optional (default=False)
+        If True the value of each internal node is printed.
+        Otherwise the value will be reported only for leaves.
+
+    show_class : bool, optional (default=None)
+        If False the class label is not printed for each node.
+        The class is printed by default for classification trees.
+        Only relevant for classification.
+
+    show_leaves_value : bool, optional (default=True)
+        If True the value on the leaves is showed.
```

Comment:
Regarding to this:

- removed the parameter `show_leaves_value` and modified the values accepted by `show_values` so that it can accept one of the following values 'only_leaves', 'all', None.
- improved documentation of `show_values` so that the difference between regression and classification is highlighted.
- the report now uses the word `value` for regression and the word `weights` for classification.